pkg/keyspace/util.go

```diff
@@ -149,3 +150,98 @@ func makeLabelRule(id uint32) *labeler.LabelRule {
 		Data:     makeKeyRanges(id),
 	}
 }
+
+// indexedHeap is a heap with index.
+type indexedHeap struct {
```

Comment:
You can check `GetAvailableKeyspaceGroupIDByKind`. Some functions of the indexed heap are indeed not used for now, if you think it's unnecessary, I can remove them.

Reply:
Saw you call indexedHeap.Top() instead of calling heap.Pop(&indexedHeap). That works too. Thanks for confirmation. Keep the unused APIs. It doesn't bother me.